lib/derivations.nix

```diff
@@ -206,4 +206,40 @@ in
   optionalDrvAttr =
     cond:
     value: if cond then value else null;
+
+  /**
+    Wrap a derivation such that instantiating it produces a warning.
+
+    # Inputs
+
+    `msg`
+    : The warning message to emit (via lib.warn).
+
+    `pkg`
+    : The derivation to wrap.
+
+    # Examples
+    :::{.example}
+    ## `lib.derivations.wrapWithWarn` usage example
+
+    ```nix
+    {
+      myPackage = wrapWithWarn "myPackage has been renamed to my-package" my-package;
+    }
+    ```
+
+    :::
+  */
+  wrapWithWarn =
+    msg: pkg:
+    pkg
+    // {
+      drvPath = lib.warn msg pkg.drvPath;
+    }
+    // builtins.listToAttrs (
```

Comment:
This reminds me of

https://github.com/NixOS/nix/blob/2e2198fd912a8725eabd96fc4530cc92401dafad/src/libexpr/primops/derivation.nix#L35

instead of adding a new default output here `pkgs.outputs or [ "out" ]` i would map over the existing list
`pkg.all` which is an empty list. 

Your current line assumes there must be some default output added called "out"

Reply:
`pkg.outputs` is the correct source of truth here.

`all` only has the values, and you can't get the output attribute name from those, or at least not without questionable code with many assumptions.

Reply:
I think this is not relevant if we just wrap (almost) all attributes, rather than specifically targetting outputs.